tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +63,98 @@ public function test_wp_enqueue_script() {
 		$this->assertSame( '', get_echo( 'wp_print_scripts' ) );
 	}
 
+	/**
+	 * Test loading strategy.
+	 *
+	 * @ticket 99999
+	 */
+	public function test_loading_strategy() {
+		/**
+		 * Async strategy testing.
+		 */
+		// Valid Async cases.
+		// No dependents, No dependencies then async.
+		wp_enqueue_script( 'main-script-a1', '/main-script-a1.js', array(), null, array( 'strategy' => 'async' ) );
+		$output   = get_echo( 'wp_print_scripts' );
+		$expected = "<script type='text/javascript' src='/main-script-a1.js' id='main-script-a1-js' async></script>\n";
+		$this->assertSame( $expected, $output );
+
+		// Invalid async cases.
+		// If any dependencies then it's not async. Since dependency is blocking(/defer) final strategy will be defer.
+		wp_enqueue_script( 'dependency-script-a2', '/dependency-script-a2.js', array(), null );
+		wp_enqueue_script( 'main-script-a2', '/main-script-a2.js', array( 'dependency-script-a2' ), null, array( 'strategy' => 'async' ) );
+		$output       = get_echo( 'wp_print_scripts' );
+		$not_expected = "<script type='text/javascript' src='/main-script-a2.js' id='main-script-a2-js' async></script>";
+		$expected     = "<script type='text/javascript' src='/main-script-a2.js' id='main-script-a2-js' defer></script>";
+		$this->assertStringNotContainsString( $not_expected, $output );
+		$this->assertStringContainsString( $expected, $output );
```

Comment:
These two assertions seem redundant. Can we do a simple `assertSame` check here to confirm the output we _do_ expect or at least only check that the string we are checking is contained? Same for the other places below where this pattern is being used.

As an aside, this type of test case, where there are multiple different cases being checking in a single test case, it's generally preferred to split these into separate methods with one assertion in each method, and name the method something descriptive, like `test_loading_strategy_with_invalid_async_registration`.

Alternately you can use a [data provider](https://docs.phpunit.de/en/9.5/writing-tests-for-phpunit.html#data-providers) to pass the parameters for different test cases.

Either way, if there is a single method performing multiple assertions, you can add an additional parameter to assertions with a string containing the message to display if it fails, like:

```
$this->assertSame( $expected, $output, 'Expected async strategy not confirmed.' );
```